packages/salesforcedx-vscode-core/src/commands/forceSourceRetrieve.ts

```diff
@@ -56,12 +57,69 @@ export class ManifestOrSourcePathGatherer
       // corresponding to the explorer path when we begin supporting multiple workspace folders
       const workspaceRootPath = vscode.workspace.workspaceFolders[0].uri.fsPath;
       const manifestPath = path.join(workspaceRootPath, 'manifest');
-      const isManifestFile = this.explorerPath.includes(manifestPath);
-      const type = isManifestFile ? FileType.Manifest : FileType.Source;
-      return { type: 'CONTINUE', data: { filePath: this.explorerPath, type } };
+      const isManifest = this.explorerPath.includes(manifestPath);
+      const type = isManifest ? FileType.Manifest : FileType.Source;
+
+      // If the type is manifest check whether or not the user selected an actual
+      // file or just the manifest directory. If the user selected the manifest
+      // directory then check that directory for manifest files. If there's only
+      // one manifest file then use it, if there none or more than one then give
+      // them a file select dialog and make them select one.
+      if (isManifest) {
+        let manifestFile = this.explorerPath;
+        const stat = fs.lstatSync(manifestFile);
+        if (stat.isDirectory()) {
+          const manifestFileList = this.getManifestFiles(manifestFile);
```

Comment:
I think that we should show an error message whenever they trigger this from the manifest folder and there's no `.xml` manifest files in it.

Reply:
Adding @ntotten and @allileong considering the discussion above revolves around always popping up a picker unless the user selected a manifest file.

Reply:
Maybe we can open up the picker as long as there exist any files or directories inside of the selected directory, but if the selected directory is empty, then notify the user that the directory does not contain any manifests.